File: soyeun/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    value = storey
    while True:
        value,remainder = divmod(value,10)
        if remainder == 0 and value == 0:
            break
        if remainder <= 4:
            answer += remainder
        elif remainder >= 6:
            answer += (10-remainder)
            value += 1
        elif remainder == 5:
            if int(str(value)[-1]) > 4:
                answer += remainder
                value += 1
            else:
                answer += remainder
    return answer

File: miri/마법의 엘리베이터.java
import java.util.*;

class Solution{
    public int solution(int storey) {
        int answer = 0;

        while ( storey > 0 ) {
            //반올림 숫자
            int upperNum = ( storey % 100 ) /10;
            //1의자리 숫자
            int num = storey % 10;

            if( num > 5 || num == 5 && upperNum >= 5) {
                answer += ( 10 - num );
                storey += 10;
            } else {
                answer += num;
            }
            storey /= 10;
        }
        return answer;
    }
}